lib/Common/DataStructures/Comparer.h

```diff
@@ -114,9 +114,11 @@ struct RecyclerPointerComparer
     }
 };
 
+// FNV-1a hash -> https://en.wikipedia.org/wiki/Fowler%E2%80%93Noll%E2%80%93Vo_hash_function
+#define CC_HASH_OFFSET_VALUE 2166136261
 #define CC_HASH_LOGIC(hash, byte) \
-    hash  = _rotl(hash, 7);        \
-    hash ^= byte;
+    hash ^= byte;                 \
+    hash *= 16777619
```

Comment:
as you mentioned, this seems to be very expensive given that we will do hash calculation every time we look/insert. Can you run browser benchmarks like Octane/speedometer to see how this impacts and is it worth pursuing?